internal/verification/authenticate_stats.go

```diff
@@ -77,6 +77,11 @@ func (v *Verifier) AuthenticateStatsToken(ctx context.Context, rawToken string)
 		if !ok {
 			return nil, fmt.Errorf("incorrect type in cache: %w", err)
 		}
+		// check that the API is enabled for this HA.
+		if !healthAuthority.EnableStatsAPI {
```

Comment:
We'll need to make a note to come back to this during productionization. This is a user error and should be 4xx not 5xx.

Reply:
this results in http.StatusUnauthorized